picard/script.py

```diff
@@ -1258,8 +1262,13 @@ def func_datetime(parser, format=None):
     # Handle case where format evaluates to ''
     if not format:
         format = '%Y-%m-%d %H:%M:%S'
-
-    return datetime.datetime.now(tz=local_tz).strftime(format)
+    try:
+        return datetime.datetime.now(tz=local_tz).strftime(format)
+    except ValueError:
+        raise ScriptRuntimeError(
+            "Unsupported format code in $%s at position %i, line %i"
+            % ('datetime', parser._x, parser._y)
```

Comment:
We could perhaps store concerned function, position and line in the `ScriptRuntimeError', so we can write:
Actual script function name can prolly be obtained from registry.

```python
raise ScriptRuntimeError("message", function, line, position)
```

Reply:
I'll have a look at this.  I think I know what you mean, but I'll need to go through the code to find out how to get the function name from the registry.  I'm not very familiar with that.

Also, should the `ScriptRuntimeError` exception be handled in `ScriptParser` or in the modules calling `ScriptParser` such as `picard.scripttofilename.script_to_filename_with_metadata()` (or even further upstream)?  Also, is there any agreement on how the exception should be handled?  For now, I suspect that it might just cause Picard to crash.  Or should the `ScriptRuntimeError` exception handling be dealt with in separate PRs?

Reply:
Is this sort of what you had in mind, or am I misunderstanding that you wanted the actual `def` function name passed to `ScriptRuntimeError` rather than the script function name?

```
    except ValueError:
        name = 'datetime'
        raise ScriptRuntimeError(
            "Unsupported format code in $%s at position %i, line %i"
            % (name, parser._x, parser._y),
            parser.functions[name].function.__name__,
            parser._x,
            parser._y
        )
```

Reply:
Yes, and no.
The idea would be to pass the short message: "Unsupported format code" along name, line and position.
And build the final message from it, something like: `"{scriptfuncname:s}:{line:d}:{position:d}: {message:s}".format(message, name, line, position)`
Perhaps we can just use hardcoded string for the script function name (simpler), but ideally it should be obtained automatically (from stack? -> python function, loop up in registry, find corresponding script file name)

About catching those: it is obvious we don't want Picard to crash, but instead display exception message to user, if it happens when the user runs the script, how/when is the question.

@phw what do you think?